services/motion/builtin/builtin.go

```diff
@@ -237,7 +258,104 @@ func (ms *builtIn) MoveOnGlobe(
 	angularVelocity float64,
 	extra map[string]interface{},
 ) (bool, error) {
-	return false, ErrNotImplemented
+	operation.CancelOtherWithLabel(ctx, builtinOpLabel)
+
+	// create a new empty framesystem which we add our base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+
+	// get the localizer from the componentName
+	localizer, ok := ms.localizer[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// assert localizer as movementSensor to get current geo point
+	movementSensor, ok := localizer.(movementsensor.MovementSensor)
+	if !ok {
+		return false, fmt.Errorf("cannot assert localizer of type %T as movementSensor", localizer)
+	}
+
+	currentGP, _, _ := movementSensor.Position(ctx, nil)
+	straightlineDistance := currentGP.GreatCircleDistance(destination)
+
+	// convert destination into spatialmath.Pose with respect to lat = 0 = lng
+	dstPose := spatialmath.GeoPointToPose(destination)
+	dstPIF := referenceframe.NewPoseInFrame(referenceframe.World, dstPose)
+
+	// convert GeoObstacles into GeometriesInFrame then into a Worldstate
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles)
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	wrldst, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+
+	// construct limits
+	limits := []referenceframe.Limit{
```

Comment:
I don't think this is correct it isn't centered around the current position

Reply:
should be now since we have the destination become relative to the currentPIF